lua/vgit/features/LiveGutter.lua

```diff
@@ -22,6 +22,27 @@ LiveGutter.hide = loop.async(function(_, buffer)
   buffer:sign_unplace()
 end)
 
+LiveGutter.store_buffer_vars = loop.async(function(_, buffer, hunks)
+  loop.await_fast_event()
+  if buffer:is_rendering() then
+    return
+  end
+  local stats_dict = {
+    added = 0,
+    changed = 0,
+    removed = 0,
+  }
+  for _, h in ipairs(hunks) do
+    -- hunk stats only contain added/removed, lines that
+    -- are both added and removed are considered "changed"
+    local changed = math.min(h.stat.added, h.stat.removed)
+    stats_dict.added = stats_dict.added + math.abs(h.stat.added - changed)
+    stats_dict.removed = stats_dict.removed + math.abs(h.stat.removed - changed)
+    stats_dict.changed = stats_dict.changed + changed
+  end
+  vim.api.nvim_buf_set_var(buffer.bufnr, 'gitsigns_status_dict', stats_dict)
```

Comment:
Can you add this as a method to the [`Buffer`](https://github.com/tanvirtin/vgit.nvim/blob/main/lua/vgit/core/Buffer.lua#L275) object. As the code base gets more complex, I find it easier to group vim api calls inside Lua objects to keep things organized.

Reply:
Absolutely, will push another change soon to #220.